### PR TITLE
Add new gossip structure for supporting repairs

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
Gossip is missing structure required for implementing fast repair 

#### Summary of Changes
Add new EpochSlots structure to gossip for supporting fast repair

Fixes #

### CODE REVIEW

File: core/src/crds_value.rs [hunk 1]
```diff
@@ -12,6 +13,58 @@ pub enum CrdsValue {
...
     Vote(Vote),
+    /// * Merge Strategy - Latest wallclock is picked
+    EpochSlots(EpochSlots),
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct EpochSlots {
+    pub from: Pubkey,
+    pub root: u64,
+    pub slots: HashSet<u64>,
```
Comment (rob-solana) [inline core/src/crds_value.rs]: is this ugly?  how is a HashSet serialized?

File: core/src/crds_value.rs [hunk 2]
```diff
@@ -94,6 +150,7 @@ impl CrdsValue {
         match self {
             CrdsValue::ContactInfo(contact_info) => contact_info.wallclock,
             CrdsValue::Vote(vote) => vote.wallclock,
+            CrdsValue::EpochSlots(vote) => vote.wallclock,
```
Comment (rob-solana) [inline core/src/crds_value.rs]: less copy-pasta, please

File: core/src/crds_value.rs [hunk 3]
```diff
@@ -102,6 +159,7 @@ impl CrdsValue {
                 CrdsValueLabel::ContactInfo(contact_info.pubkey())
             }
             CrdsValue::Vote(vote) => CrdsValueLabel::Vote(vote.pubkey()),
+            CrdsValue::EpochSlots(slots) => CrdsValueLabel::EpochSlots(slots.pubkey()),
```
Comment (rob-solana) [inline core/src/crds_value.rs]: looks like we need a CrdsValue trait?

File: core/src/crds_value.rs [hunk 4]
```diff
@@ -175,6 +244,7 @@ mod test {
             match v {
                 CrdsValueLabel::ContactInfo(_) => hits[0] = true,
                 CrdsValueLabel::Vote(_) => hits[1] = true,
+                CrdsValueLabel::EpochSlots(_) => hits[1] = true,
```
Comment (rob-solana) [inline core/src/crds_value.rs]: hits[1]?

Comment (rob-solana) [reply core/src/crds_value.rs]: want to grow hits to size 3 and use hits[2]?

**Comment by sagar-solana** [comment]
Yeah i can refractor gossip at some point. Each crds value should carry a gossip meta. Signature, wall clock, from id.

File: core/src/crds_value.rs (hunk 4 above)
Comment (carllin) [reply core/src/crds_value.rs]: oops, yeah good catch

File: core/src/crds_value.rs (hunk 1 above)
Comment (carllin) [reply core/src/crds_value.rs]: It should be efficient, the Serialize trait is implemented for HashSet, and should map pretty closely to a vector according to docs: https://serde.rs/data-model.html

**Pull request merged**